cli/bpmetadata/parser/state_parser_test.go

```diff
@@ -0,0 +1,180 @@
+package parser
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func TestParseOutputTypesFromState_WithSimpleTypes(t *testing.T) {
+	t.Parallel()
+	stateData := []byte(`
+{
+  "version": 4,
+  "terraform_version": "0.12.0",
+  "serial": 8,
+  "lineage": "491fd8f4-81b5-9890-520c-8a173c36e483",
+  "outputs": {
+    "boolean_output": {
+      "value": true,
+      "type": "bool"
+    },
+    "number_output": {
+      "value": 42,
+      "type": "number"
+    },
+    "string_output": {
+      "value": "foo",
+      "type": "string"
+    }
+  },
+  "resources": []
+}
+`)
+	want := map[string]*structpb.Value{
+		"boolean_output": structpb.NewStringValue("bool"),
+		"number_output":  structpb.NewStringValue("number"),
+		"string_output":  structpb.NewStringValue("string"),
+	}
+	got, err := ParseOutputTypesFromState(stateData)
+	if err != nil {
+		t.Errorf("ParseOutputTypesFromState() error = %v", err)
+		return
+	}
+	if diff := cmp.Diff(got, want, cmp.Comparer(compareStructpbValues)); diff != "" {
+		t.Errorf("ParseOutputTypesFromState() mismatch (-got +want):\n%s", diff)
+	}
+}
+
+func TestParseOutputTypesFromState_WithComplexTypes(t *testing.T) {
+	t.Parallel()
+	stateData := []byte(`
+{
+  "version": 4,
+  "terraform_version": "0.12.0",
+  "serial": 8,
+  "lineage": "491fd8f4-81b5-9890-520c-8a173c36e483",
+  "outputs": {
+    "interpolated_deep": {
+      "value": {
+        "foo": "bar",
+        "map": {
+          "bar": "baz",
+          "id": "424881806176056736"
+        },
+        "number": 42
+      },
+      "type": [
+        "object",
+        {
+          "foo": "string",
+          "map": [
+            "object",
+            {
+              "bar": "string",
+              "id": "string"
+            }
+          ],
+          "number": "number"
+        }
+      ]
+    },
+    "list": {
```

Comment:
nit: let's not use the type as output variable name! 
```suggestion
    "list_output": {
```

Reply:
Good catch!